scripts/unlinked_artists.coffee

```diff
@@ -2,52 +2,75 @@
 
-db.articles.find (err, articles) ->
+#Pull all of the articles into memory and extract out their html
+db.articles.find({ published: true }).toArray (err, articles) ->
   return exit err if err
   debug "Found #{articles?.length} articles, extracting html..."
-  articlesHtmls = articles.map getHtmlFrom
-  db.artists.find (err, artists) ->
-    return exit err if err
-    debug "Found #{artists?.length} artists..."
-    articlesWithUnlinkedNames = _.compact _.flatten(
-      for article, i in articles
-        unlinkedArtistSlugs = findUnlinked artists, articlesHtmls[i]
-        debug "Searched #{i + 1 * 1000} articles..." if i % 1000 is 0
-        if unlinkedArtistSlugs.length
-          { id: article._id, slugs: unlinkedArtistSlugs }
-        else
-          null
-    )
-    debug "DONE"
-    for data in articlesWithUnlinkedNames
-      debug "Article #{data.id} is missing links to #{data.slugs.join ', '}"
-    process.exit()
+  articlesElements = articles.map getHtmlFrom
+
+  # Parse the HTML and find any instances of artist names not inside <a>s
+  csv = [['Artist Name', 'Unlinked Articles', 'Unlinked Artsy URLs', 'Total Articles Unlinked', 'Times Unmentioned']]
+  csvHash = {}
+  # {
+  #   "Andy Warhol": {
+  #     unlinkedArticles: ['writer.artsy.net/...']
+  #     unlinkedArtsyUrls: ['artsy.net/article/...']
+  #     totalUnlinked: 5
+  #     timesUnmentioned: 10
+  #   }
+  # }
+  for article, i in articles
+    unlinkedArtistNamesAndCounts = findUnlinked articlesElements[i]
+    for artistName, unmentionedCount of unlinkedArtistNamesAndCounts
+      csvHash[artistName] ?= { unlinkedArticles: [], unlinkedArtsyUrls: [], totalUnlinked: 0, timesUnmentioned: 0 }
+      csvHash[artistName].unlinkedArticles.push(
+        "http://writer.artsy.net/articles/#{article._id}/edit"
+      )
+      csvHash[artistName].unlinkedArtsyUrls.push(
+        "http://artsy.net/article/#{article.slugs.slice(-1)[0]}"
+      )
+      csvHash[artistName].timesUnmentioned += unmentionedCount
+    debug "Searched #{i + 1 * 1000} articles..." if i % 1000 is 0
+  for artistName, data of csvHash
+    csv.push [
+      artistName
+      data.unlinkedArticles.join(' | ')
+      data.unlinkedArtsyUrls.join(' | ')
+      data.unlinkedArticles.length
+      data.timesUnmentioned
+    ]
+
+  # Build up a CSV string and write the file
+  csvstring = ''
+  csvstring += row.join(',') + '\n' for row in csv
+  fs.writeFileSync(__dirname + '/tmp/unlinked_artists.csv', csvstring)
+  debug 'CSV written'
+  process.exit()
 
-findUnlinked = (artists, articlesHtml) ->
-  slugs = []
-  for artist, i in artists
-    name = (artist.first?.trim() or '') + ' ' + (artist.last?.trim() or '')
-    if name.trim() and articlesHtml.indexOf(name) >= 0
-      slugs.push artist._slugs[0]
-  slugs
+findUnlinked = ($) ->
+  namesAndCounts = {}
+  html = $.html().toLowerCase()
+  for name in ARTISTS
+    nameLink = 'artsy.net/artist/' + name.split(' ').join('-')
```

Comment:
I think you will want to do `name.toLowerCase().split(' ').join('-')`. However, this won't be as reliable as testing if any of the links have the artist name inside of them because some slugs aren't 1:1 with the name e.g. there might be slugs like `andy-warhol1`. However, I think this good enough and will match most of the time.

The other way might look something like...

````coffeescript
linkExists = name.toLowerCase() in $('a').map(-> $(this).text().toLowerCase().trim())
````